libavcodec/jpeg2000dec.c

```diff
@@ -408,6 +418,75 @@ static int get_siz(Jpeg2000DecoderContext *s)
     s->avctx->bits_per_raw_sample = s->precision;
     return 0;
 }
+/* get extended capabilities (CAP) marker segment */
+static int get_cap(Jpeg2000DecoderContext *s, Jpeg2000CodingStyle *c)
+{
+    uint32_t Pcap;
+    uint16_t Ccap_i[32] = { 0 };
+    uint16_t Ccap_15;
+    uint8_t P;
+
+    if (bytestream2_get_bytes_left(&s->g) < 6) {
+        av_log(s->avctx, AV_LOG_ERROR, "Insufficient space for CAP\n");
+        return AVERROR_INVALIDDATA;
+    }
+
+    Pcap = bytestream2_get_be32u(&s->g);
+    for (int i = 0; i < 32; i++) {
+        if ((Pcap >> (31 - i)) & 1)
+            Ccap_i[i] = bytestream2_get_be16u(&s->g);
+    }
+    Ccap_15 = Ccap_i[14];
+    if (!Ccap_15) {
+        av_log(s->avctx, AV_LOG_ERROR, "Wrong Ccap_15 value\n");
+        return AVERROR_INVALIDDATA;
+    }
+    s->isHT = 1;
+    av_log(s->avctx, AV_LOG_INFO, "This is an HTJ2K codestream.\n");
+    // Bits 14-15
+    switch ((Ccap_15 >> 14) & 0x3) {
+        case 0x3:
+            s->Ccap15_b14_15 = HTJ2K_MIXED;
+            break;
+        case 0x1:
+            s->Ccap15_b14_15 = HTJ2K_HTDECLARED;
+            break;
+        case 0x0:
+            s->Ccap15_b14_15 = HTJ2K_HTONLY;
+            break;
+        default:
+            av_log(s->avctx, AV_LOG_ERROR, "Bits 14-15 of Ccap_15 is wrong.\n");
```

Comment:
```suggestion
            av_log(s->avctx, AV_LOG_ERROR, "Unknown CCap value.\n");
```

Also, maybe this should be a warning and we should try to decode.